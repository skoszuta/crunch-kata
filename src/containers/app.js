import React from 'react';
import { connect } from 'react-redux';
import styled from 'styled-components';

import { fetchVariablesWithOrder } from 'actions/variables-actions';
import VariableCatalog from 'components/variable-catalog';
import VariablePositionFinder from 'containers/variable-position-finder';
import VariableFinder from 'containers/variable-finder';

const ContentWrapper = styled.div`
  width: 100%;
  padding: 1rem;
  
  @media (min-width: 600px) {
    padding: 2rem;
  }
`;

const Heading = styled.h1`
  margin: 0 0 1rem 0;
`;

const Section = styled.section`
  margin-top: 2rem;
  
  &:first-child {
    margin-top: 0;
  }
`;

export class App extends React.Component {
  componentDidMount() {
    this.props.fetchVariablesWithOrder();
  }

  render() {
    if (this.props.variables === null) {
      return (
        <div>Loading...</div>
      );
    }

    return (
      <ContentWrapper>
        <Section>
          <Heading>The component</Heading>
          <VariableCatalog variables={this.props.variables}/>
        </Section>
        <Section>
          <Heading>Variable position finder</Heading>
          <VariablePositionFinder/>
        </Section>
        <Section>
          <Heading>Variable finder</Heading>
          <VariableFinder/>
        </Section>
      </ContentWrapper>
    );
  }
}

function mapStateToProps({ variables }) {
  return { variables };
}

export default connect(mapStateToProps, { fetchVariablesWithOrder })(App);
