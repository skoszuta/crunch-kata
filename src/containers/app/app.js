import React from 'react';
import { connect } from 'react-redux';
import styled from 'styled-components';

import { fetchVariablesWithOrder } from '../../actions/variables-actions';
import VariableCatalog from '../../components/variable-catalog/variable-catalog';
import VariablePositionFinder from '../variable-position-finder/variable-position-finder';

const ContentWrapper = styled.div`
  width: 100%;
  margin: 0 auto;
  font-family: Helvetica, Arial, sans-serif;
  
  @media (min-width: 600px) {
    width: 60%;
  }
  
  @media (min-width: 960px) {
    width: 40%;
  }
`;

class App extends React.Component {
  componentDidMount() {
    this.props.fetchVariablesWithOrder();
  }

  render() {
    if (this.props.variables === null) {
      return (
        <div>Loading...</div>
      );
    }

    return (
      <ContentWrapper>
        <section>
          <h1>The component</h1>
          <VariableCatalog variables={this.props.variables}/>
        </section>
        <section>
          <h1>Variable position finder</h1>
          <VariablePositionFinder/>
        </section>
        <section>
          <h1>Variable finder</h1>
        </section>
      </ContentWrapper>
    );
  }
}

function mapStateToProps({ variables }) {
  return { variables };
}

export default connect(mapStateToProps, { fetchVariablesWithOrder })(App);
