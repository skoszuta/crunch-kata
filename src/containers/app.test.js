import React from 'react';
import { shallow, render } from 'enzyme';
import 'jest-styled-components';

import { App } from './app';

describe('App', () => {
  const fetchVariablesWithOrder = jest.fn();
  const variablesLoaded = { index: {}, order: [] };

  afterEach(() => {
    fetchVariablesWithOrder.mockReset();
  });

  it('should render a loading indicator', () => {
    const wrapper = render(
      <App variables={null} fetchVariablesWithOrder={fetchVariablesWithOrder} />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('should render correctly with variables loaded', () => {
    const wrapper = shallow(
      <App variables={variablesLoaded} fetchVariablesWithOrder={fetchVariablesWithOrder} />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('should dispatch fetchVariablesWithOrder on mount', () => {
    shallow(
      <App variables={null} fetchVariablesWithOrder={fetchVariablesWithOrder} />
    );

    expect(fetchVariablesWithOrder).toBeCalled();
  });
});
