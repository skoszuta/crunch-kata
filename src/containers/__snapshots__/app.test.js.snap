// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render a loading indicator 1`] = `
<div>
  Loading...
</div>
`;

exports[`App should render correctly with variables loaded 1`] = `
<styled.div>
  <styled.section>
    <styled.h1>
      The component
    </styled.h1>
    <VariableCatalog
      variables={
        Object {
          "index": Object {},
          "order": Array [],
        }
      }
    />
  </styled.section>
  <styled.section>
    <styled.h1>
      Variable position finder
    </styled.h1>
    <Connect(VariablePositionFinder) />
  </styled.section>
  <styled.section>
    <styled.h1>
      Variable finder
    </styled.h1>
    <Connect(VariableFinder) />
  </styled.section>
</styled.div>
`;
